client/ayon_core/pipeline/publish/lib.py

```diff
@@ -933,3 +934,54 @@ def get_publish_instance_families(instance):
         families.discard(family)
     output.extend(families)
     return output
+
+
+def get_instance_expected_output_path(
+        instance: pyblish.api.Instance,
+        representation_name: str,
+        ext: Optional[str] = None,
```

Comment:
Why is extension optional?
```suggestion
        ext: str,
```

Reply:
It's because it's also optional in the `representation` dict data structure in `instance.data["representations"]`
```python
representation = {
    "name": "usd",
    ...
}
representation = {
    "name": "usd",
    "ext": "usda",
    ...
}
```
Both of these are valid in instance data (unfortunately?)

Reply:
Let me know if you have follow up questions on this - otherwise I'll mark it resolved.

Reply:
I think both arguments should be mandatory. You can keep the check if not ext use name, but the argument itself should not be optional.

Reply:
Implemented with https://github.com/ynput/ayon-core/pull/295/commits/379d61082971bb19963a4af6bd9110a91e1bca32